### PR TITLE
adding to_bin_str function

### PR DESCRIPTION
**Pull request by MSPDUTTA**
Added to_bin_str function and two tests

### CODE REVIEW

File: uint.h [hunk 1]
```diff
@@ -516,6 +516,23 @@ class UInt {
     }
     os << std::dec;
   }
+  
+public:
+   std::string to_bin_str(std::ostream& os) const {
```
Comment (sbeamer) [inline uint.h]: Perhaps this should take no arguments

File: tester.cc [hunk 2]
```diff
@@ -23,6 +26,8 @@ SInt<80>  b80s("0xefbe8ae0d38ab7f36dda");
 SInt<128> a128s("0x6e0939370acc19daec06e9c13db50674");
 SInt<128> b128s("0xbeb828fdbac591dba8e38eeb433f563d");
 
+UInt<192> abcdu("0xffffffffffffffff0000000000000000ff1f1f1f1f1f1f1f");
```
Comment (sbeamer) [inline tester.cc]: Could you use the existing test variables (e.g. a16u, b64u, etc...) instead of declaring new ones?

File: tester.cc [hunk 3]
```diff
@@ -3,7 +3,10 @@
 
 #include "uint.h"
 #include "sint.h"
-
+#include <iostream>
+#include <fstream>
```
Comment (sbeamer) [inline tester.cc]: I don't believe firrtl-sig needs to deal with files (including following lines).

**Review by sbeamer — CHANGES_REQUESTED** [review]

File: tester.cc (hunk 2 above)
Comment (MSPDUTTA) [reply tester.cc]: Yes

**Review by sbeamer — APPROVED** [review]

**Pull request merged**